pkg/apiserver-impl/starterserver.go

```diff
@@ -0,0 +1,57 @@
+package apiserver_impl
+
+import (
+	"context"
+	"fmt"
+	openapi "github.com/redhat-developer/odo/pkg/apiserver-gen/go"
+	"github.com/redhat-developer/odo/pkg/state"
+	"github.com/redhat-developer/odo/pkg/util"
+	"k8s.io/klog"
+	"net/http"
+)
+
+func StartServer(ctx context.Context, cancelFunc context.CancelFunc, port int, stateClient state.Client) {
```

Comment:
I think you could create an interface with a method instead of this function, so we can inject it in the clientset, and mock it for testing.

This way, we could unit-test the CLI layer by checking the startserver method is called when the --apiserver flag is passed only.